core/src/main/scala/cats/syntax/applicativeError.scala

```diff
@@ -34,4 +34,7 @@ final class ApplicativeErrorOps[F[_], E, A](val fa: F[A]) extends AnyVal {
 
   def recoverWith(pf: PartialFunction[E, F[A]])(implicit F: ApplicativeError[F, E]): F[A] =
     F.recoverWith(fa)(pf)
+
+  def onError(fa: F[A])(pf: PartialFunction[E, F[Unit]])(implicit F: ApplicativeError[F, E]): F[A] =
+    F.onError(fa)(pf)
```

Comment:
I assumed that the discipline tests provides enough coverage for the implementation, no? Anyway I am not against adding a `EitherT[Writer,...` to the unit tests